Code/GraphMol/ChemTransforms/MolFragmenter.cpp

```diff
@@ -595,4 +595,301 @@ ROMol *fragmentOnBRICSBonds(const ROMol &mol) {
+    
+    // Mark the original order of the nbr atoms including the dummy
+    //  The goal is to copy the dummy chiral order over to the
+    //  atom being bonded
+    void mark(Atom *chiral_atom, Atom *dummy_atom, Atom *new_atom) const {
+        if(chiral_atom->getChiralTag()) {
+            std::string mark = "__molzip_mark_" + std::to_string(chiral_atom->getIdx());
+            chiral_atom->setProp("__molzip_chiral_mark", mark);
+            int order = 0;
+            auto &m = chiral_atom->getOwningMol();
+            for(auto nbrIdx : boost::make_iterator_range(m.getAtomNeighbors(chiral_atom))) {
+                m[nbrIdx]->setProp<int>(mark, order);
+                ++order;
+            }
+            new_atom->setProp<int>(mark, dummy_atom->getProp<int>(mark));
+        }
+        
+        // check bond stereo
+        auto &m = chiral_atom->getOwningMol();
+        auto bond  = m.getBondBetweenAtoms(chiral_atom->getIdx(), dummy_atom->getIdx());
+        for(auto nbrIdx : boost::make_iterator_range(m.getAtomNeighbors(chiral_atom))) {
+            auto bond  = m.getBondBetweenAtoms(chiral_atom->getIdx(), nbrIdx);
+            if(bond->getStereo()) {
+                std::string mark = "__molzip_bond_stereo_mark";
+                std::vector<Atom*> atoms;
+                bool has_dummy = false;
+                for(auto idx : bond->getStereoAtoms()) {
+                    if(idx == dummy_atom->getIdx()) {
+                        atoms.push_back(new_atom);
+                        has_dummy = true;
+                    } else {
+                        atoms.push_back(m[idx]);
+                    }
+                }
+                if(has_dummy) {
+                    bond->setProp<std::vector<Atom*>>(mark, atoms);
+                    bond->setProp<Bond::BondStereo>("__molzip_bond_stereo", bond->getStereo());
+                }
+            }
+        }
+    }
+    
+    // Backup the original chirality
+    void mark_chirality() const {
+        PRECONDITION(a, "Must have a begin atom to bond");
+        PRECONDITION(b, "Must have an end atom to bond");
+        PRECONDITION(a_dummy, "Must have a begin dummy atom");
+        PRECONDITION(b_dummy, "Must have an end dummy atom");
+        
+        mark(a, a_dummy, b);
+        mark(b, b_dummy, a);
+    }
+    
+    // bond a<->b for now only use single bonds
+    //  XXX FIX ME take the highest bond order.
+    bool bond(RWMol &newmol) const {
+        if (!a || !b) {
+            BOOST_LOG(rdWarningLog) << "Incomplete atom labelling, cannot make bond" << std::endl;
+            return false;
+        }
+        if (!a->getOwningMol().getBondBetweenAtoms(a->getIdx(), b->getIdx())) {
+            assert (&a->getOwningMol() == &newmol);
+            auto bond_type_a = newmol.getBondBetweenAtoms(a->getIdx(), a_dummy->getIdx())->getBondType();
+            auto bond_type_b = newmol.getBondBetweenAtoms(b->getIdx(), b_dummy->getIdx())->getBondType();
```

Comment:
Are you really 100% sure that these bonds will always be here?
If not, please check that the two `getBondBetweenAtoms()` actually return non-null